pyispyb/app/extensions/auth/token.py

```diff
@@ -23,8 +33,10 @@ def generate_token(username, groups, permissions):
 
     token = jwt.encode(
         {
-            "username": username,
-            "groups": groups,
+            "login": login,
+            "personId": personId,
+            "givenName": givenName,
```

Comment:
Do we need `givenName` and `familyName` in the token? I think `login` and `personId` are usefull as they can be used in all DB requests to authorize the user. Names are nice to have in the return type for display in the UI, but I'm not sure it is necessary in the token?

Reply:
The idea is it works with https://github.com/ispyb/py-ispyb/pull/201/files#diff-3ed2a7780185b0d1a5963842a5c643399dd501a214c1a132a5e6e65197ac0cc3R77 without needing to make a new query to `Person`. We could not provide this info here, and instead requery `Person` in `GET /user/current` if you prefer?

Reply:
My idea was to store in the token only the information that we need to have for most requests, to avoid making an extra DB querry every time (so that's permissions for basic authorization, and person id and/or username for authorization through proposal/session). 
The token is exchanged on every requests so I guess it is best to keep it as short as possible. Of course having these parameters too doesn't make it huge so it is completely fine either way, but they are only useful for this route that should not be called as much as the others.